sw/device/silicon_creator/mask_rom/mask_rom.c

```diff
@@ -106,6 +106,11 @@ static rom_error_t mask_rom_init(void) {
   // Configure UART0 as stdout.
   uart_init(kUartNCOValue);
 
+  // There are not conditional checks before writing to this CSR because it is
+  // expected that if relevant Ibex countermeasures are disabled, this will
+  // result in a nop.
```

Comment:
@vogelpi, @GregAC  is this assumption correct? Thanks

Reply:
This is correct, I just checked the RTL. If dummy instructions are disabled (SecureIbex parameter), the write is just ignored but no illegal instruction exception is triggered.